File: front/src/models/dtos/availability/createAvailabilityDto.ts
import { Address } from '@/models/entities/address';

/**
 * DTO for creating a new availability.
 */
export type CreateAvailabilityDto = {
  /**
   * The start date of the availability.
   * Formatted as an ISO 8601 string.
   */
  startDate: string;

  /**
   * The end date of the availability.
   * Formatted as an ISO 8601 string.
   */
  endDate: string;

  /**
   * The address of the availability.
   */
  address: Address;

  /**
   * The id of the job category of the availability.
   */
  jobCategoryId: string;

  /**
   * The range around the address (in kilometers).
   */
  range: number;
};

File: backend/api_gateway/src/main/kotlin/com/linkedout/backend/dto/availability/CreateAvailabilityDto.kt
package com.linkedout.backend.dto.availability

import com.linkedout.backend.model.Address
import java.time.LocalDate

data class CreateAvailabilityDto(
    val startDate: LocalDate,
    val endDate: LocalDate,
    val address: Address,
    val range: Int,
    val jobCategoryId: String
)
